### PR TITLE
Testing and coverage

### LINKED ISSUE #3: How to add configuration of Unit test cases - Karma-ui5, Qunit
Please provide a guide about adding and running test cases of Qunit, Karma-UI5 in the typescript project. Presently getting below error since the project codebase is in 'webapp' folder - 
Error: Could not find application directory of project oaui: /<path-to-project>/oaui/webapp
Even if provide a path of 'src' folder tests are not executing.

### PR DESCRIPTION
**Pull request by petermuessig**
Ingredients:

* Functional tests using TypeScript QUnit tests
* Integration tests using TypeScript OPA tests (a bit hacky still)
* Test automation using Karma
* Code coverage support using Istanbul

🚀  Good to go now, PR#19 has been merged into `main`, `testing` has been rebased on `main` and this PR has been put on top of the `testing` branch

Fixes #3

### CODE REVIEW

**Comment by petermuessig** [comment]
Hi @akudev , hi @margopolo ,

I also extended the `ui5-typescript-helloworld` project with the steps to enable for testing and also code coverage. This would now be really a complete guide. I also fixed the `ui5-tooling-transpile` extension to work out of the box with `babel-plugin-istanbul` which is now also a great addition and simplifies the code coverage enablement a lot! On Stack@SAP this topic has been discussed and a bit cumbersome workarounds have been made - one works only with the classic Babel step to transpile from src to webapp and instrument there - and the other is quite tricky with an extra Babel run in Karma - but as the UI5 Tooling + `ui5-tooling-transpile` runs nevertheless in the Karma run as middleware I only had to correct the proper path to be used for the instrumentation and gotcha! it worked without any strange workaround anymore... This looks really awesome now - I think the main question is the OPA stuff - would we be ok with a workaround or should we just show the QUnit tests? BTW: QUnit tests - @akudev - even QUnit itself recommends the Promise.all workaround: https://api.qunitjs.com/config/autostart/ - the main problem why it doesn't work with the `Promise.all(...).then` is because of the following code in the Babel transpilation: https://github.com/ui5-community/babel-plugin-transform-modules-ui5/blob/9c7d26e6aeebc77f5d8faec941fbd994b77de222/packages/plugin/src/utils/templates.js#L87 - we can also consider to get rid of the `reject` here in case of a module doesn't export anything - which could be indeed the case. Let's follow-up on this extra... With `Promise.all(...).finally` it works great!

**Comment by petermuessig** [comment]
🚀 **Ready for Review** @akudev 

Once the PR https://github.com/ui5-community/babel-plugin-transform-modules-ui5/pull/92 has been reviewed and accepted by @codeworrior we can also adopt the `Promise.finally(...)` in the testsuites to `Promise.then(...)` - but the rest looks good now!

From my POV, we can merge this into the testing branch now. I anyhow rebased the testing branch on the latest state of the main branch. @akudev, I saved your Babel setup in the testing-BabelSetup branch - it's you decision what to do with it...

File: step-by-step.md [hunk 1]
```diff
@@ -0,0 +1,708 @@
+# A Detailed Guide to Create a UI5 TypeScript App From Scratch in Five to Ten Steps
```
Comment (akudev) [inline step-by-step.md:1]: Hm, the idea was not to repeat the basic setup in this branch. Can we like build on top of the setup in the main branch and only add the testing here?

File: webapp/test/integration/pages/App.ts [hunk 2]
```diff
@@ -1,59 +1,59 @@
...
-export class AppPageActions extends Opa5 {
-	and: AppPageActions // TODO: would be nice if this would not need to be defined
-
-	iPressTheSayHelloWithDialogButton() {
-		return this.waitFor({
-			id: "helloButton",
-			viewName,
-			actions: new Press(),
-			errorMessage: "Did not find the 'Say Hello With Dialog' button on the App view"
-		}) // TODO: would be nice if this cast would not be needed
```
Comment (akudev) [inline webapp/test/integration/pages/App.ts:16]: Ummm... which cast? I swear there was one when I wrote that. :-) Anyway, a) good you added the cast, but b) bad you removed the comment: actually it was meant to be one of the hints what should be improved with regards to TS OPA support.

File: webapp/test/testsuite.qunit.html [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-		<script src="testsuite.qunit.js" data-sap-ui-testsuite></script>
-	</head>
-	<body>
-	</body>
-</html>
\ No newline at end of file
+<!DOCTYPE html>
+<html>
+<head>
+	<title>QUnit test suite for Worklist</title>
```
Comment (akudev) [inline webapp/test/testsuite.qunit.html]: worklist?

**Review by akudev — APPROVED** [review]

File: webapp/test/testsuite.qunit.html (hunk 3 above)
Comment (petermuessig) [reply webapp/test/testsuite.qunit.html]: Copy paste issue... 😄

File: webapp/test/integration/pages/App.ts (hunk 2 above)
Comment (petermuessig) [reply webapp/test/integration/pages/App.ts:16]: Todo will be added

File: step-by-step.md (hunk 1 above)
Comment (petermuessig) [reply step-by-step.md:1]: We can discuss separetly - finally I would like to have one comprehensive guide from 0 to hero

**Comment by petermuessig** [comment]
You can compare the latest diff here: https://github.com/SAP-samples/ui5-typescript-helloworld/compare/eb4982ebd920e9b0b1efa6d0e3f59413dd34575f..f45c13d1771eb492cc4540872835b4194a57edce

Should all be good to go now... As said, ideally this can even go into the master/main branch in future...

**Pull request merged**